### PR TITLE
style: Move font-variant-numeric outside of mako

### LINKED ISSUE #19276: Move font-variant-numeric outside of mako.
I'll take this one as a sub-issue of #19015

### PR DESCRIPTION
**Pull request by CYBAI**
This is a sub-PR of #19015 
r? emilio 

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #19276 
- [x] These changes do not require tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19277)

### CODE REVIEW

File: components/style/values/specified/font.rs [hunk 1]
```diff
@@ -1350,6 +1350,229 @@ impl Parse for FontVariantLigatures {
...
+        write_value!(VariantNumeric::SLASHED_ZERO => "slashed-zero");
+        write_value!(VariantNumeric::ORDINAL => "ordinal");
+
+        debug_assert!(has_any);
+        Ok(())
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl_gecko_keyword_conversions!(VariantNumeric, u8);
```
Comment (CYBAI) [inline components/style/values/specified/font.rs]: @emilio Previously, in the PRs of `font-variant-east-asian` and `font-variant-ligatures`, I copied the content of `impl_gecko_keyword_conversions` macro. However, they will be looked like very duplicate codes.
Thus, if using macro is better, I'd like to send another PR to implement this macro for `font-variant-east-asian` and `font-variant-ligatures`.

Comment (emilio) [reply components/style/values/specified/font.rs]: Yup, that sounds great!

**Review by emilio — APPROVED** [review]
Wow, you're on a roll @CYBAI \o/

Thanks so much, this looks great!

File: components/style/values/specified/font.rs (hunk 1 above)
Comment (CYBAI) [reply components/style/values/specified/font.rs]: Great! Thanks; I'll send a PR for it!
